src/coreclr/vm/dllimport.cpp

```diff
@@ -2252,31 +2242,25 @@ DWORD NDirectStubLinker::EmitProfilerBeginTransitionCallback(ILCodeStream* pcsEm
 {
     STANDARD_VM_CONTRACT;
 
-    if (SF_IsForwardDelegateStub(dwStubFlags) || SF_IsCALLIStub(dwStubFlags))
-    {
-        // secret argument does not contain MD nor UMEntryThunk
-        pcsEmit->EmitLoadNullPtr();
-    }
-    else
+    if (!SF_IsForwardDelegateStub(dwStubFlags) && !SF_IsCALLIStub(dwStubFlags))
```

Comment:
Are the changes in this method because the things emitted were incorrect previously, or is it somehow chained from the changing the ICall into a QCall?

Reply:
Good question. These changes here are because I extracted logic from the previous FCall and moved it into the generated stub. This was @jkotas's suggestion at https://github.com/dotnet/runtime/pull/106604#discussion_r1722263111. Previously it was in two places and I moved it to three (3). However, Jan pointed out I could move it to one in the actual stub.